client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
```

Comment:
I still think handling `co.uk.` is more important than some super rare domains. But if you are feeling strongly towards handling it this way I think that's alright as well.